bindings/node/lib/common.js

```diff
@@ -91,7 +91,29 @@ function promiseOrCallback(callback, fn) {
   });
 }
 
+/**
+ * Ask for KMS credentials.
+ *
+ * This returns anything that looks like the kmsProviders original input
+ * option. It can be empty, and any provider specified here will override
+ * the original ones.
+ *
+ * @param {AutoEncrypter|ClientEncryption} encrypter The encrypter.
+ */
+async function askForKMSCredentials(encrypter) {
+  // First attempt to use the user provided callback to get the credentials.
+  const creds = encrypter._onKmsProviderRefresh ? encrypter._onKmsProviderRefresh() : {};
+  // If the credentials are empty and we have an additional callback provided
+  // by the driver to attempt to fetch them, attempt to use that.
+  if (!creds.aws && encrypter._onEmptyKmsProviders) {
```

Comment:
This currently checks for `aws` specifically but could be made more generic in the future for other KMS providers.

Reply:
I guess it might be worth adding a comment as to _why_ we single out AWS for now?

Reply:
Note on AWS added.